src/Type/Plugin/Connection/PluginConnectionDefinition.php

```diff
@@ -26,9 +26,13 @@ class PluginConnectionDefinition {
 	 * @return mixed
 	 * @since 0.0.5
 	 */
-	public static function connection() {
+	public static function connection( $from_type = 'Root' ) {
 
 		if ( null === self::$connection ) {
+			self::$connection = [];
```

Comment:
Rather than doing this, wouldn't it make sense to have `self::$connection` have a default value of `[]` and then just check if it's empty like you are on like 35?